src/plugins/auto/plugin.cpp

```diff
@@ -806,7 +789,42 @@ std::string MultiDeviceInferencePlugin::GetDeviceList(const std::map<std::string
 
     return allDevices;
 }
-
+std::vector<std::string > MultiDeviceInferencePlugin::ParsePrioritiesDevices(const std::string& priorities, const char separator) {
```

Comment:
Why not use MetaParseDevice directly?

Reply:
deadlock happened when using `parseMetaDevice()` to check if device name in the priorities is available. call trace as below.
```
Core ie;
ie->compile_model(network, "AUTO", configs);
ie->GetCPPpluginByName("AUTO");
hold the pluginMutex and create AUTO plugin.
AUTO plugin::SetConfig(configs);
AUTO Plugin::CheckConfig();
AUTO Plugin::parseMetaDevices(ov::device::priorities.value());
GetCore()->GetAvaiableDevices();
ie->GetListOfDevicesInRegistry();
hold the pluginMutex again and deadlock happened here.
```